ivi-layermanagement-examples/multi-touch-viewer/src/multi-touch-viewer.c

```diff
@@ -641,39 +643,65 @@ touch_event_test_main(struct touch_event_test_params *p_params)
     return 0;
 }
 
-void
+static void
 usage(int status)
 {
-    printf("usage: multi-touch-viewer [OPTION]\n");
-    printf("  -p : print received touch point\n");
+    fprintf(stderr, "    -h,  --help              display this help and exit.\n"
+                    "    -p,  --print-log         print received touch point.\n"
+                    "    -s,  --surface-type      select the type of surface, default is the ivi_app_surface:\n"
+                    "                              - wl_shell: to using the wl_shell_surface.\n"
+                    "                              - ivi_app: to using the ivi_app_surface.\n");
     exit(status);
 }
 
-int
-main(int argc, char **argv)
+static void 
+parse_options(int argc, char *argv[])
 {
-    _UNUSED_(argc);
-    _UNUSED_(argv);
-    struct touch_event_test_params params;
+    int opt = -1, option_index = 0;
+    static const struct option options[] = {
+        { "help", no_argument, NULL, 'h' },
+        { "print-log", no_argument, NULL, 'p' },
+        { "surface-type", required_argument, NULL, 's' },
+        { 0, 0, NULL, 0 }
+    };
 
-    memset(&params, 0x00, sizeof params);
+    while (1) {
+        opt = getopt_long(argc, argv, "hps:", options, NULL);
 
-    if (argc == 2)
-    {
-        if (0 == strcmp(argv[1], "-p"))
-        {
-            g_is_print_log = 1;
-        }
-        else
-        {
-            usage(EXIT_SUCCESS);
+        if (opt == -1)
+            break;
+
+        switch (opt) {
+            case 'h':
+                usage(0);
+                break;
+            case 'p':
+                g_is_print_log = 1;
+                break;
+            case 's':
+                if (strcmp(optarg, "wl_shell") == 0)
+                    g_surface_type = WL_SHELL_SURFACE;
+                break;
+            default:
+                usage(-1);
+                break;
         }
     }
+    printf("multi-touch-viewer runs with %s surface and %s log\n", 
```

Comment:
Check on the grammar please. The state will be printed as e.g.:

multi-touch-viewer runs with wl_shell surface and enable log

This is grammatically in correct. Shall we change this is to: multi-touch viewer runs with wl_shell and logging enabled. ?

Reply:
Thank you for mention this, I updated here and commit message